coil-base/src/main/java/coil/map/ResourceUriMapper.kt

```diff
@@ -0,0 +1,12 @@
+package coil.map
+
+import android.content.ContentResolver
+import android.net.Uri
+import androidx.annotation.DrawableRes
+
+class ResourceUriMapper : Mapper<Uri, @DrawableRes Int> {
+
+    override fun handles(data: Uri) = data.scheme == ContentResolver.SCHEME_ANDROID_RESOURCE
+
+    override fun map(data: Uri) = checkNotNull(data.lastPathSegment?.toIntOrNull()) { "Malformed resource Uri: $data" }
```

Comment:
Hi @colinrtwhite 
Please provide support for another type of Uri pattern as well.

<img width="1101" alt="uri_format_exception" src="https://user-images.githubusercontent.com/20519350/63928716-ab8e8180-ca6d-11e9-82d5-12f69e477d49.png">

Reply:
@pkpdeveloper Good point! I'll add support for this.